### PR TITLE
[ENG-7094][eas-build-job] improve job image validation

### PR DESCRIPTION
**Pull request by szdziedzic**
# Why

https://linear.app/expo/issue/ENG-7094/better-error-handling-when-image-is-valid-but-doesnt-exist-on

We want to validate images by taking resource class into account. We want to move image sanitization logic to the `turtle-v2` repo so we don't have to publish `eas-build-job` every time that new image is added.

# How

Only validate if the `image` value is string type. Remove all other logic and transfer it to `turtle-v2`.

# Test Plan

Automated tests. Test with Turtle changes locally.

### CODE REVIEW

File: packages/eas-build-job/src/job.ts [hunk 1]
```diff
@@ -13,24 +13,38 @@ interface ImageMatchArgs {
   workflow: Workflow;
 }
 
-export const JobSchema = Joi.object<Job>({
+export const JobPlatfomSchema = Joi.object<Job>({
```
Comment (dsokal) [inline packages/eas-build-job/src/job.ts:16]:
```suggestion
export const JobPlatformSchema = Joi.object<Job>({
```

Comment (dsokal) [reply packages/eas-build-job/src/job.ts:16]: Also, can you verify this is not used elsewhere? E.g. in other repos.

File: packages/eas-build-job/src/job.ts [hunk 2]
```diff
@@ -13,24 +13,38 @@ interface ImageMatchArgs {
...
-  .when(Joi.object({ platform: Platform.ANDROID }).unknown(), { then: Android.JobSchema })
-  .when(Joi.object({ platform: Platform.IOS }).unknown(), { then: Ios.JobSchema });
+});
 
-export function sanitizeJob(
-  rawJob: object,
-  { sdkVersion, reactNativeVersion }: { reactNativeVersion?: string; sdkVersion?: string } = {}
-): Job {
-  const { value, error } = JobSchema.validate(rawJob, {
+function validateJobPlatform(rawJob: object): Platform {
```
Comment (dsokal) [inline packages/eas-build-job/src/job.ts:22]:
```suggestion
function sanitizeJobPlatform(rawJob: object): Platform {
```

Comment (dsokal) [reply packages/eas-build-job/src/job.ts:22]: Usually: 
- `validate...` only validates the input, can throw an error, but does not return anything
- `sanitize...` validates the input, can throw an error, and returns the validated (and potentially changed) input

**Review by wkozyra95 — CHANGES_REQUESTED** [review]
I don't think we should have anything related to the resource class in this repo, validation should happen in turtle-api

**Review by dsokal — APPROVED** [review]

**Pull request merged**